RocketChat/Rocket.Chat
# [FIX] Empty DMs go to sidebar's top after updating user preferences

## Proposed changes (including videos or screenshots)


- Do not use `_updatedAt` as `lm` field when merging info from subscriptions and rooms (since this field is updated along with the user preferences).



## Issue(s)

Fix #23617

## Steps to test or reproduce

1. Create a new direct message and send a message in another room (so that the DM is not shown on top of the list);
2. Go to **My Account > Preferences > Notifications** and update Desktop/Push/Email notification setting (any of them);
3. Return to the main screen and check the sidebar.
**_Current behavior:_** empty DMs go to the top of the list.
**_Expected behavior:_** empty DMs should now stay at the bottom of the list.

## Further comments

The `lastRoomUpdate` variable is used as a backup field to the `lm` field when it is not available in the room (that is, when no message has been sent in the room). Changing user notification preferences also updates the `_updatedAt` field in every subscription, which implies that empty DMs will be sorted to top when the sidebar is sorted. Instead, we could keep this field as undefined so that they're sorted to the bottom of the list.

TC-79
